cloud/amazon/rds_v2.py

```diff
@@ -0,0 +1,1142 @@
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            e = get_exception()
+            module.fail_json(msg=str(e))
+
+        if module.params.get('wait'):
+            status_achieved, response = wait_for_status(client, module, 'available')
+            if not status_achieved:
+                module.fail_json(msg='Error waiting for RDS instance creation - please check the AWS console')
+
+    result = response
+    return changed, result
+
+
+def reboot_db_instance(client, module):
+    changed = False
+    required_vars = ['db_instance_identifier']
+    valid_vars = ['force_failover']
+
+    params = validate_parameters(required_vars, valid_vars, module)
+
+    try:
+        response = client.reboot_db_instance(**params)
+        changed = True
+    except botocore.exceptions.ClientError as e:
+        e = get_exception()
+        module.fail_json(msg=str(e))
+
+    if module.params.get('wait'):
+        status_achieved, response = wait_for_status(client, module, 'available')
+        if not status_achieved:
+            module.fail_json(msg='Error waiting for RDS instance creation - please check the AWS console')
+
+    result = response
+    return changed, result
+
+
+def replicate_db_instance(client, module):
+    changed = False
+    required_vars = ['db_instance_identifier', 'source_db_instance_identifier']
+    valid_vars = ['db_instance_class', 'availability_zone', 'port', 'auto_minor_version_upgrade', 'iops',
+                  'option_group_name', 'publicly_accessible', 'tags', 'db_subnet_group_name', 'storage_type',
+                  'copy_tags_to_snapshot', 'monitoring_interval']
+
+    params = validate_parameters(required_vars, valid_vars, module)
+
+    # Check if instance already exists. If so, do nothing and return the instance
+    db_instance = get_db_instance(client, module)
+    if db_instance:
+        response = db_instance
+        changed=False
+    else:
+        try:
+            response = client.create_db_instance_read_replica(**params)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            e = get_exception()
+            module.fail_json(msg=str(e))
+
+        if module.params.get('wait'):
+            status_achieved, response = wait_for_status(client, module, 'available')
+            if not status_achieved:
+                module.fail_json(msg='Error waiting for RDS replica creation - please check the AWS console')
+
+    result = response
+    return changed, result
+
+
+def promote_db_instance(client, module):
+    changed = False
+    required_vars = ['db_instance_identifier']
+    valid_vars = ['backup_retention_period', 'preferred_backup_window']
+    instance_name = module.params.get('db_instance_identifier')
+
+    params = validate_parameters(required_vars, valid_vars, module)
+
+    #check if instance already exists. If so, do nothing and return the instance
```

Comment:
I think this comment is invalid, since if the instance does exist the module promotes it